docs/core/deploying/native-aot/fixing-warnings.md

```diff
@@ -0,0 +1,80 @@
+
+## Reacting to AOT warnings
+
+The AOT warnings are meant to bring predictability to native AOT builds. Majority of AOT warning are about possible run time exception in situations when native code wasn't generated to support the scenario. The broadest category is `RequiresDynamicCodeAttribute`.
+
+### RequiresDynamicCode
+
+<xref:System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute> is simple and broad: it's an attribute that means the member has been annotated incompatible with AOT, meaning that it might use reflection or some other mechanism to create new native code at runtime. This attribute is used when code is fundamentally not AOT compatible, or the native dependency is too complex to statically predict at build time. This would often be true for methods that use the Type.MakeGenericType API, reflection emit, or other runtime code generation technologies. An example would be:
+
+```csharp
+[RequiresDynamicCode("Use 'MethodFriendlyToAot' instead")]
+void MethodWithReflectionEmit() { ... }
+
+void TestMethod()
+{
+    // IL3050: Using method 'MethodWithReflectionEmit' which has 'RequiresDynamicCodeAttribute'
+    // can break functionality when AOT compiling. Use 'MethodFriendlyToAot' instead.
+    MethodWithReflectionEmit();
+}
+```
+
+There aren't many workarounds for `RequiresDynamicCode`. The best fix is to avoid calling the method at all when building as native AOT and use something else that's AOT-compatible. If you're writing a library and it's not in your control whether or not to call the method, you can also add `RequiresDynamicCode` to your own method. This will annotate your method as not AOT compatible. Adding `RequiresDynamicCode` will silence all AOT warnings in the given method, but will produce a warning whenever someone else calls it. For this reason, it is mostly useful to library authors to "bubble up" the warning to a public API.
+
+If you can somehow determine that the call is safe, and all native code will be available at runtime, you can also suppress the warning using <xref:System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute>. For example:
+
+```csharp
+[RequiresDynamicCode("Use 'MethodFriendlyToAot' instead")]
+void MethodWithReflectionEmit() { ... }
+
+[UnconditionalSuppressMessage("Aot", "IL3050:RequiresDynamicCode",
+    Justification = "The unfriendly method is not reachable with AOT")]
+void TestMethod()
+{
+    If (RuntimeFeature.IsDynamicCodeSupported)
+        MethodWithReflectionEmit(); // warning suppressed
+}
+```
+
+`UnconditionalSuppressMessage` is like `SuppressMessage` but it can be seen by `publish` and other post-build tools. `SuppressMessage` and `#pragma` directives are only present in source so they can't be used to silence warnings from the build. Be very careful when suppressing AOT warnings: it's possible that the call may be AOT-compatible now, but as you change your code that may change, and you may forget to review all the suppressions.
```

Comment:
```suggestion
`UnconditionalSuppressMessage` is like `SuppressMessage` but it can be seen by `publish` and other post-build tools. `SuppressMessage` and `#pragma` directives are only present in source, so they can't be used to silence warnings from the build.

> [!CAUTION]
> Be careful when suppressing AOT warnings. The call might be AOT-compatible now, but as you update your code, that might change, and you might forget to review all the suppressions.
```